BuildAssets/repos/test-apt.sh

```diff
@@ -1,26 +1,41 @@
 #!/bin/bash
 
-# This script requires environment variables to run:
-#   S3_PUBLISH_ENDPOINT, OCTOPUS_CLI_SERVER, OCTOPUS_CLI_API_KEY, OCTOPUS_SPACE, OCTOPUS_EXPECT_ENV
+if [[ -z "$PUBLISH_LINUX_EXTERNAL" || -z "$OCTOPUS_CLI_SERVER" || -z "$OCTOPUS_CLI_API_KEY" \
+   || -z "$OCTOPUS_SPACE" || -z "$OCTOPUS_EXPECT_ENV" ]]; then
+  echo -e 'This script requires the following environment variables to be set:
+  PUBLISH_LINUX_EXTERNAL, OCTOPUS_CLI_SERVER, OCTOPUS_CLI_API_KEY, OCTOPUS_SPACE, OCTOPUS_EXPECT_ENV' >&2
+  exit 1
+fi
 
-# Configure apt
+if [[ "$PUBLISH_LINUX_EXTERNAL" == "true" ]]; then
+  ORIGIN="https://apt.octopus.com"
+else
+  ORIGIN="http://prerelease.apt.octopus.com"
+fi
+
+# Configure docker environment
 export DEBIAN_FRONTEND=noninteractive
 apt-get update --quiet 2 || exit
 apt-get install --no-install-recommends --yes apt-utils >/dev/null 2>&1 || exit # silence debconf warnings
-apt-get install --no-install-recommends --yes gnupg curl software-properties-common >/dev/null || exit
-apt-key adv --fetch-keys "http://$S3_PUBLISH_ENDPOINT/public.key" 2>&1 || exit
-add-apt-repository "deb http://$S3_PUBLISH_ENDPOINT/ stretch main" || exit
+
+# The following should be kept in sync with our documented download instructions (but more automated and quieter).
+
+echo "## Configuring apt"
+DIST=$({ grep --perl --no-filename --only-matching --no-messages \
+  '^deb\s+(\[[^\]#]*?\]\s+)?[^\s#]+(debian|ubuntu)[^\s#]*\s+\K\w+' /etc/apt/sources.list /etc/apt/sources.list.d/* \
+  | sort | uniq --count | sort --reverse --numeric; echo 0 stable; } | awk '{ print $2; exit; }')
+apt-get install --no-install-recommends --yes gnupg curl ca-certificates apt-transport-https 2>&1 >/dev/null || exit
+curl --silent --show-error --fail --location "$ORIGIN/public.key" | APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key add - 2>&1 || exit
+echo "deb $ORIGIN/ $DIST main" > /etc/apt/sources.list.d/octopus.com.list || exit
 apt-get update --quiet 2 || exit
 
-# Install
+echo "## Installing tentacle, octopuscli"
 apt-get install --no-install-recommends --yes tentacle octopuscli >/dev/null || exit
 
-# Test
-echo "== Testing Tentacle =="
+echo "## Testing Tentacle"
```

Comment:
How does this cross over with Tentacle? (ie, are we going to run the same tests over there? how are we keeping the scripts in sync?)